From ca94ad5a631dcd9e91f8f3e32c65a88d5f6f64d0 Mon Sep 17 00:00:00 2001
From: Bruno Pagani <bruno.n.pagani@gmail.com>
Date: Mon, 27 Jun 2022 10:38:33 +0000
Subject: [PATCH] Fix name visibility issues for LLVM 14 build

Ensure that `DEBUG_TYPE` is defined properly whenever `LLVM_DEBUG` is used.
It must be declared after all LLVM headers have been included (`llvm/Support/Debug.h` among these).
---
 src/xe/GlobalsLocalization.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/xe/GlobalsLocalization.cpp b/src/xe/GlobalsLocalization.cpp
index e623e3141..8762d0ae6 100644
--- a/src/xe/GlobalsLocalization.cpp
+++ b/src/xe/GlobalsLocalization.cpp
@@ -36,7 +36,6 @@
 */
 
 #ifdef ISPC_XE_ENABLED
-#define DEBUG_TYPE "localize_globals"
 
 #include "GlobalsLocalization.h"
 
@@ -71,6 +70,8 @@
 #include <llvm/Support/raw_ostream.h>
 #include <llvm/Transforms/Scalar.h>
 
+#define DEBUG_TYPE "localize_globals"
+
 using namespace llvm;
 
 using LocalizationLimitT = int32_t;
